iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp

```diff
@@ -403,13 +431,11 @@ class IOX_NO_DISCARD expected<ValueType, ErrorType>
     ///         return someErrorProneFunction();
     ///     }
     /// @endcode
+    //
     template <typename T>
+    // NOLINTNEXTLINE(hicpp-explicit-conversions) see doxygen brief section
     operator expected<T>() const noexcept;
 
-    /// @brief conversion operator to an optional.
-    /// @return optional containing the value if the expected contains a value, otherwise a nullopt
-    operator optional<ValueType>() const noexcept;
-
```

Comment:
Do we need to add this to the release notes as breaking change? ... I think I have to do the same for the Duration operator timeval